product_to_bizerba_scale/models/product_template.py

```diff
@@ -0,0 +1,145 @@
+        string='Scale Tare Weight',
+        help="Set here Constant tare weight"
+        " for the given product. This tare will be substracted when"
+        " the product is weighted. Usefull only for weightable product.\n"
+        "The tare is defined with kg uom."
+    )
+
+    # View Section
+    @api.multi
+    def send_scale_create(self):
+        for product in self:
+            # TODO: Should check if the product has a scale group
+            product._send_to_scale_bizerba('create', True)
+        return True
+
+    @api.multi
+    def send_scale_write(self):
+        for product in self:
+            # TODO: Should check if the product has a scale group
+            product._send_to_scale_bizerba('write', True)
+        return True
+
+    @api.multi
+    def send_scale_unlink(self):
+        for product in self:
+            # TODO: Should check if the product has a scale group
+            product._send_to_scale_bizerba('unlink')
+        return True
+
+    # Custom Section
+    def _send_to_scale_bizerba(self, action, send_product_image=False):
+        log_obj = self.env['product.scale.log']
+        log_obj.create({
+            'log_date': fields.Datetime.now(),
+            'scale_system_id': self.scale_group_id.scale_system_id.id,
+            'product_id': self.id,
+            'action': action,
+            'send_product_image': send_product_image,
+        })
+
+    def _check_vals_scale_bizerba(self, vals):
+        system = self.scale_group_id.scale_system_id
+        system_fields = [x.name for x in system.field_ids]
+        for product_field in ADDITIONAL_FIELDS:
+            if product_field not in system_fields:
+                system_fields.append(product_field)
+        vals_fields = list(vals.keys())
+        return set(system_fields).intersection(vals_fields)
+
+    def is_in_scale(self):
+        """Return True if the current product should be in the scale
+        system.
+        """
+        self.ensure_one()
+        return self.active and self.sale_ok and self.scale_group_id
+
+    def is_new_in_scale(self, vals):
+        """Return True if the current product will be new in the scale
+        system after the write.
+        """
+        return not self.is_in_scale() and self.will_be_in_scale(vals)
+
+    def will_be_in_scale(self, vals):
+        """Return True if the current product will be in the scale
+        system after the write.
+        """
+        self.ensure_one()
+        return (vals.get('active', self.active)
+                and vals.get('sale_ok', self.sale_ok)
+                and vals.get('scale_group_id', self.scale_group_id))
+
+    # Overload Section
+    @api.model
+    def create(self, vals):
+        product = super(ProductTemplate, self).create(vals)
+        if product.is_in_scale():
+            product._send_to_scale_bizerba('create')
+        return product
+
+    @api.multi
+    def write(self, vals):
+        defered = {}
```

Comment:
I have made a type for `defered`, it should be spelled `deferred`. I will be nice if you can fix that. :)

Reply:
renamed